### DISCUSSION TITLE
issue while using I2S

### DISCUSSION THREAD

**Discussion by jalmince** [Bodmer/TFT_eSPI#2703] [Q&A - General]
Hi Bodmer, and congrats for your job !

Thanks to your lib I wrote code which runs fine on the LilyGo T-embed ( using its embeded mic and speaker, while displaying animated sprites)

Now I am trying to port it onto the LilyGo T-Display-S3.

Therefore I connected an I2S mic and an I2S sound boards to the T-Display-S3.
( code and connections are ok, since  I can listen to a web radio, or read the mic samples.)

Issue is that when I add the TFT-eSPI, the displaying is at best very slow, and most of the time interrupted.

I have stripped down my code ( below  ) to the bare essentials showing my issue:

the code "flashes" the display by  alternating tft.fillScreen( TFT_BLACK) and tft.fillScreen( TFT_RED) in the loop.
of course it works absolutely fine.

But if I just try to install the I2S driver for the mic, ( without even reading the I2S), "flashing" is gone ...

I suspect a conflict, but ive struggled for days now, and can't find a solution.

code below if you have time to help ....

cheers,
Jean-Paul

```
#include "Arduino.h"
#include "TFT_eSPI.h" /* Please use the TFT library provided in the library. */
#include "pin_config.h"
#include <driver/i2s.h>

TFT_eSPI tft = TFT_eSPI();

// ---------------------------------------- I2S for microphone
// Connections to INMP441 
#define I2S_WS  1
#define I2S_SD  2
#define I2S_SCK 3
#define I2S_PORT_MIC I2S_NUM_1  

void install_I2S_mic() {
  #define I2S_CH_MIC I2S_NUM_1
  #define SAMPLE_RATE 16000

    i2s_config_t i2s_config = {
      .mode = (i2s_mode_t)(I2S_MODE_MASTER | I2S_MODE_RX ),
      .sample_rate = SAMPLE_RATE,
      .bits_per_sample = I2S_BITS_PER_SAMPLE_16BIT,
      .channel_format = I2S_CHANNEL_FMT_ALL_LEFT,
      .communication_format = I2S_COMM_FORMAT_STAND_I2S,
      .intr_alloc_flags = ESP_INTR_FLAG_LEVEL1,
      .dma_buf_count = 8,
      .dma_buf_len = 64,
      .use_apll = false,
      .tx_desc_auto_clear = false,
      .fixed_mclk = 0,
      .mclk_multiple = I2S_MCLK_MULTIPLE_256,  
      .bits_per_chan = I2S_BITS_PER_CHAN_16BIT,
      .chan_mask = (i2s_channel_t)(I2S_TDM_ACTIVE_CH0 | I2S_TDM_ACTIVE_CH1),};
    i2s_pin_config_t i2s_mic_pins = {
      .bck_io_num = I2S_SCK,
      .ws_io_num = I2S_WS,
      .data_out_num = I2S_PIN_NO_CHANGE, 
      .data_in_num = I2S_SD};
    i2s_driver_install(I2S_CH_MIC, &i2s_config, 0, NULL);   
    i2s_set_pin(I2S_CH_MIC, &i2s_mic_pins);
    i2s_zero_dma_buffer(I2S_CH_MIC);
}

void setup() 
{
    Serial.begin(115200);
    delay(3000);
    tft.begin();
    // install_I2S_mic() ;  if this one  is commented out, the tft functions get impaired !!
} 

void loop(){
  tft.fillScreen( TFT_BLACK);
  delay( 500); 
  tft.fillScreen( TFT_RED);
  delay( 500);  
  Serial.print(".") ;
}
```

**Comment by Bodmer** [comment]
I suspect the I2S interface is consuming processor power. You might try allocating I2S audio processing to the second processor. I don't think this is a TFT_eSPI library issue.

**Comment by ham0nd** [comment]
i also meet your problems, when i init my i2s, my tft program will crash,have you solved your problems? @jalmince
